From bd933fa03d8e4d4a243a5491e68dcffd77d883e2 Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Fri, 29 Mar 2024 10:16:36 +0530
Subject: [PATCH] drivers: mtd: nand: Updating part MX35UF2GE4AD-Z4I

The MX35UF2GE4AD-Z4I nand part has support for 8-bit
ECC, 2Gb flash density and 128 byes of OOB. Density of
MX35UF4GE4AD-Z4I nand part is 4Gb. Updating the
correct info in the nand id table.

Change-Id: I6d218384dc40e4f8a72c0b572e2c61d5c74c3ae2
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 drivers/mtd/nand/raw/nand_ids.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/raw/nand_ids.c b/drivers/mtd/nand/raw/nand_ids.c
index 5b6de79b7102..f9336bee3900 100644
--- a/drivers/mtd/nand/raw/nand_ids.c
+++ b/drivers/mtd/nand/raw/nand_ids.c
@@ -79,13 +79,13 @@ struct nand_flash_dev nand_flash_ids[] = {
 	{"MT29F1G01ABBFDWB-IT SPI NAND 1G 1.8V 4-bit",
 		{ .id = {0x2c, 0x15} },
 		SZ_2K, SZ_128, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512) },
-	{"MX35UF4GE4AD-Z4I SPI NAND 1G 1.8V",
+	{"MX35UF4GE4AD-Z4I SPI NAND 4G 1.8V",
 		{ .id = {0xc2, 0xb7} },
 		SZ_4K, SZ_512, SZ_256K, 0, 2, 256, NAND_ECC_INFO(8, SZ_512)},
 
 	{"MX35UF2GE4AD-Z4I SPI NAND 2G 1.8V",
 		{ .id = {0xc2, 0xa6} },
-		SZ_2K, SZ_128, SZ_128K, 0, 2, 160, NAND_ECC_INFO(4, SZ_512)},
+		SZ_2K, SZ_256, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512)},
 
 	{"GD5F1GQ5REYIH SPI NAND 1G 1.8V",
 		{ .id = {0xc8, 0x21} },
-- 
2.34.1

